rules/file-contents.js

```diff
@@ -29,30 +29,124 @@ async function fileContents(fs, options, not = false) {
     return new Result(
       'Did not find file matching the specified patterns',
       fileList.map(f => {
-        return { passed: false, pattern: f }
+        return { passed: !options['fail-on-non-existent'], pattern: f }
```

Comment:
Ideally this would be in the commit with the test fix. But not going to beat on that topic.